src/providers/splash.ts

```diff
@@ -0,0 +1,53 @@
+import { AmountWithDecimal, Provider } from "@/types/core";
+import { AmmPool, makeNativePools, Pools } from "@ergolabs/ergo-dex-sdk";
+import { AssetAmount, Explorer, RustModule } from "@ergolabs/ergo-sdk";
+
+class Splash implements Provider {
+  private baseUrl: string;
+
+  constructor(url: string) {
+    this.baseUrl = url;
+  }
+
+  async getOrderBook(marketId: string) {
+    try {
+      const response = await fetch(
+        `${this.baseUrl}/platform-api/v1/trading-view/order-book?base=${marketId}&quote=.`,
+        {
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json",
+          },
+        }
+      );
+      return response.json();
+    } catch (error) {
+      console.error("Error fetching order book:", error);
```

Comment:
It's not a good practice to catch an error only to log it, and then re-throw it. You can simply remove the `try/catch` block.

Reply:
Ok. I delete it.